dolphin/processor/config.py

```diff
@@ -490,15 +482,24 @@ def get_index_lens_light_model_list(self):
          (for multiple filters)
         """
         if 'lens_light' in self.settings['model']:
-            index_lens_light_model_list = []
-            index_num = 0
-            for i in range(self.band_number):
-                single_index_list = []
-                for j in range(len(self.settings['model']['lens_light'])):
-                    single_index_list.append(index_num)
-                    index_num += 1
-                index_lens_light_model_list.append(single_index_list)
-            return index_lens_light_model_list
+            if self.band_number == 1:
+                index_list = [[]]
+                for k, model in enumerate(
+                        self.settings['model']['lens_light']):
+                    index_list[0].append(k)
+                return index_list
+            else:
+                if 'lens_light_band_index' in self.settings['model']:
+                    index_list_settings = \
+                       self.settings['model']['lens_light_band_index']
+                    index_list = [[] for _ in range(self.band_number)]
+                    for i, model in enumerate(index_list_settings):
+                        index_list[model].append(i)
+                    for k in index_list:
+                        assert k != [], "One of the bands have no lens light"
+                    return index_list
+                else:
+                    raise ValueError('Missing lens_light_band_index')
```

Comment:
Add more description to the error message as `'Missing "lens_light_band_indices" in the settings file!'`